tests/test_binary_classification_stats.py

```diff
@@ -0,0 +1,104 @@
+    def setUp(self):
+        self.binary_classification_stats = BinaryClassificationStats()
+
+    def test_remove_relevant_ranks(self):
+        self.assertEqual(
+            self.binary_classification_stats.remove_relevant_ranks(disease_results, [2, 4]), [1, 2]
+        )
+
+    def test_add_classification_for_known_entities_true_pos(self):
+        self.binary_classification_stats.add_classification_for_known_entities([1])
+        self.assertEqual(
+            self.binary_classification_stats,
+            BinaryClassificationStats(
+                true_positives=1, true_negatives=0, false_positives=0, false_negatives=0
+            ),
+        )
+
+    def test_add_classification_for_known_entities_false_neg(self):
+        self.binary_classification_stats.add_classification_for_known_entities([3])
+        self.assertEqual(
+            self.binary_classification_stats,
+            BinaryClassificationStats(
+                true_positives=0, true_negatives=0, false_positives=0, false_negatives=1
+            ),
+        )
+
+    def test_add_classification_for_known_entities_multiple_results(self):
+        self.binary_classification_stats.add_classification_for_known_entities([1, 3, 100])
+        self.assertEqual(
+            self.binary_classification_stats,
+            BinaryClassificationStats(
+                true_positives=1, true_negatives=0, false_positives=0, false_negatives=2
+            ),
+        )
+
+    def test_add_classification_for_other_entities_false_pos(self):
+        self.binary_classification_stats.add_classification_for_other_entities([1])
+        self.assertEqual(
+            self.binary_classification_stats,
+            BinaryClassificationStats(
+                true_positives=0, true_negatives=0, false_positives=1, false_negatives=0
+            ),
+        )
+
+    def test_add_classification_for_other_entities_true_neg(self):
+        self.binary_classification_stats.add_classification_for_other_entities([10])
+        self.assertEqual(
+            self.binary_classification_stats,
+            BinaryClassificationStats(
+                true_positives=0, true_negatives=1, false_positives=0, false_negatives=0
+            ),
+        )
+
+    def test_add_classification_for_other_entities_multiple_results(self):
+        self.binary_classification_stats.add_classification_for_other_entities([1, 2, 3, 4, 5])
+        self.assertEqual(
+            self.binary_classification_stats,
+            BinaryClassificationStats(
+                true_positives=0, true_negatives=4, false_positives=1, false_negatives=0
+            ),
+        )
+
+    def test_add_classification(self):
+        self.binary_classification_stats.add_classification(disease_results, [1, 2])
+        self.assertEqual(
+            self.binary_classification_stats,
+            BinaryClassificationStats(
+                true_positives=1, true_negatives=2, false_positives=0, false_negatives=1
```

Comment:
Given 2 results, with ranks 1, 2, assuming the rank 1 is a TP, then 2 must be a TN and then the FP and FN should be 0, no?

Reply:
No, for this I imagined that there are 2 variants linked for a sample and one of the variants is found ranked 1 in the results and the other is ranked 2. This would mean there is 1 TP and 1 FN. There are 2 true negatives as these are the other results that are not relevant.